### DISCUSSION TITLE
Accessing DispatchRuntime.ChannelDispatcher.ErrorHandlers in IContractBehaviour hook

### DISCUSSION THREAD

**Discussion by DublinDel** [CoreWCF/CoreWCF#312] [Q&A]
Hi all

First of all congrats on the progress to date.

I have to add to the ErrorHandlers collection on the ChannelDispatcher through the ApplyDispatchBehavior implementation of IContractBehavior.

Currently the access to ChannelDispatcher is internal on the DispatchRuntime class. Can this be made public or an method added to add to the ErrorHandlers collection from the DispatchRuntime object?

This is what I have:

`        
public void ApplyDispatchBehavior(ContractDescription contractDescription, ServiceEndpoint endpoint, DispatchRuntime dispatchRuntime)
        {
            dispatchRuntime.ChannelDispatcher.ErrorHandlers.Add(new LogErrorBehaviourAttribute());
        }
`
Or is this all simplified in the IApplicationBuilder configuration in the Core bootstrapping?

Regards
Derek

**Comment by birojnayak** [comment]
Hi Derek,
We discussed, instead of making it public inside ChannelDispatcher , you could use```dispatchRuntime.EndpointDispatcher.ChannelDispatcher.ErrorHandlers``` ... let me know if that helps... if you want more reasons, let me know can add more details...

**Reply by DublinDel** [reply]
Hi Biroj

That's perfect for me. 

I will be looking at stripping out as much of the error handling from WCF anyway and using .NET Core fault handling where possible. These are just small legacy issues I am facing at this particular moment.

Thank you all for your help.

Regards
Derek
